nav2_util/src/robot_utils.cpp

```diff
@@ -143,6 +143,66 @@ bool getTransform(
   return true;
 }
 
+bool getTransform(
+  const std::string & source_frame_id,
+  const std::string & target_frame_id,
+  const tf2::Duration & transform_tolerance,
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  geometry_msgs::msg::TransformStamped & transform_stamped)
+{
+  geometry_msgs::msg::TransformStamped transform;
+
+  if (source_frame_id == target_frame_id) {
+    // We are already in required frame
+    return true;
+  }
+
+  try {
+    // Obtaining the transform to get data from source to target frame
```

Comment:
Could this be called by the nonstamped version above so we reduce duplicate code?